src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2460,6 +2487,14 @@ void processContextElement
       searchContextProviders(tenant, servicePathV, *enP, ceP->contextAttributeVector, cerP);
       cerP->statusCode.fill(SccOk);
       responseP->contextElementResponseVector.push_back(cerP);
+
+      //
+      // If no context providers found, then the UPDATE was simply for a non-found entity and an error should be returned
+      //
+      if (forwardsPending(responseP) == false)
```

Comment:
Yes, this is the change.
The 'why' is pretty clear ...
This is an UPDATE => could be forwarded to context providers.
If forwarded (found in ngsi9 registration) then it cannot be a 404 Not Found, but if there are no forwards to be made (and nothing local found) then we have a 404 to report back to the caller.
I believe this was an old bug and it is now fixed ...